src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpCodeParser.cs

```diff
@@ -1553,6 +1553,13 @@ private void HandleDirective(DirectiveDescriptor descriptor)
                         outputKind = SpanKind.Code;
                         break;
                     case DirectiveTokenKind.String:
+                        if (!At(CSharpSymbolType.StringLiteral))
+                        {
+                            Context.ErrorSink.OnError(
+                                CurrentStart,
+                                LegacyResources.FormatDirectiveExpectsQuotedStringLiteral(descriptor.Name),
+                                CurrentSymbol.Content.Length);
+                        }
                         AcceptUntil(CSharpSymbolType.WhiteSpace, CSharpSymbolType.NewLine);
```

Comment:
Hmm, if that's what we want to do. Lets be specific and move the `AcceptUntil` into the error case (with a comment) and have an else case with `AcceptAndMoveNext`